scripts/quests/otherAreas/The_Sand_Charm.lua

```diff
@@ -0,0 +1,143 @@
+
+                    if zPos <= 29 or zPos >= 38 or xPos <= 16 or xPos >= 32 then
+                        if quest:getVar(player, 'Prog') == 0 then
+                            return quest:progressEvent(125) -- I know he's out there
+                        elseif quest:getVar(player, 'Prog') == 2 then
+                            return quest:progressEvent(124) -- Celestina doesn't want to hear it anymore
+                        end
+                    end
+                end,
+            },
+
+            ['Zexu'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(123) -- Word is, pirates got 'im
+                    end
+                end,
+            },
+
+            ['Celestina'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return quest:progressEvent(126, xi.item.SAND_CHARM)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [123] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+
+                [124] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [125] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+
+                [126] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Celestina'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHas(trade, xi.item.SAND_CHARM) then
+                        return quest:progressEvent(127, 0, xi.item.SAND_CHARM)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [127] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:tradeComplete()
+                        player:setCharVar('SmallDialogByBlandine', 1)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_COMPLETED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Blandine'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getCharVar('SmallDialogByBlandine') == 1 then
+                        return quest:event(128)
```

Comment:
Since we want this to take priority - `progressEvent(128)` I would imagine would be more beneficial here.